### PR TITLE
[Localization] Localize Berry Heal message

### PR DESCRIPTION
**Pull request by EnochG1**
## What are the changes?
Add the option to localize message text when Berry heals HP or PP.

## Why am I doing these changes?
All text should be localizable.

## What did change?
 - Change the code for localize > src/data/berry.ts
 - Added localization keys in each language folders. > battle.ts
 - Change English message when Berry heals HP (sync with original game.)
 - Translated them to Korean.

#### En
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

#### Fr
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

#### De
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

#### Pt_BR
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

#### Zh_CN
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

#### Zh_TW
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

#### Ko
<details>
<summary>Click to expand/collapse</summary>

| Case | Result |
| :-----: | :-----: |
| HP Heal | image |
| PP Heal | image |

</details>

## How to test the changes?
Manually (using overrides.ts)
(example)
```Typescript
export const STARTING_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "BERRY", count: 1, type: BerryType.LEPPA}];
export const OPP_HELD_ITEMS_OVERRIDE: Array<ModifierOverride> = [ {name: "BERRY", count: 1, type: BerryType.SITRUS}];
export const MOVESET_OVERRIDE: Array<Moves> = [ Moves.FALSE_SWIPE, Moves.RECOVER ];
```

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/locales/de/battle.ts [hunk 1]
```diff
@@ -68,6 +68,8 @@ export const battle: SimpleTranslationEntries = {
   "drainMessage": "{{pokemonName}} wurde Energie abgesaugt",
   "regainHealth": "KP von {{pokemonName}} wurden wieder aufgefrischt!",
   "stealEatBerry": "{{pokemonName}} stole and ate\n{{targetName}}'s {{berryName}}!",
+  "ppHealBerry": "{{pokemonNameWithAffix}} restored PP to its move {{moveName}}\nusing its {{berryName}}!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:71]:
```suggestion
  "ppHealBerry": "{{berryName}} von {{pokemonNameWithAffix}} füllt AP von {{moveName}} auf!",
```

File: src/locales/de/battle.ts [hunk 2]
```diff
@@ -68,6 +68,8 @@ export const battle: SimpleTranslationEntries = {
...
+  "hpHealBerry": "{{pokemonNameWithAffix}} restored its health using\nits {{berryName}}!",
```
Comment (CodeTappert) [inline src/locales/de/battle.ts:72]:
```suggestion
  "hpHealBerry": "{{berryName}} füllt KP von {{pokemonNameWithAffix}} auf!",
```

File: src/locales/zh_CN/battle.ts [hunk 3]
```diff
@@ -62,6 +62,8 @@ export const battle: SimpleTranslationEntries = {
   "skipItemQuestion": "你确定要跳过拾取道具吗？",
   "eggHatching": "咦？",
   "stealEatBerry": "{{pokemonName}} stole and ate\n{{targetName}}'s {{berryName}}!",
+  "ppHealBerry": "{{pokemonNameWithAffix}} restored PP to its move {{moveName}}\nusing its {{berryName}}!",
+  "hpHealBerry": "{{pokemonNameWithAffix}} restored its health using\nits {{berryName}}!",
```
Comment (mercurius-00) [inline src/locales/zh_CN/battle.ts:66]:
```suggestion
  "ppHealBerry": "{{pokemonNameWithAffix}}用{{berryName}}\n回复了{{moveName}}的PP！",
  "hpHealBerry": "{{pokemonNameWithAffix}}用{{berryName}}\n回复了体力！",
```

File: src/locales/zh_TW/battle.ts [hunk 4]
```diff
@@ -54,6 +54,8 @@ export const battle: SimpleTranslationEntries = {
   "escapeVerbSwitch": "切換",
   "escapeVerbFlee": "逃跑",
   "stealEatBerry": "{{pokemonName}} stole and ate\n{{targetName}}'s {{berryName}}!",
+  "ppHealBerry": "{{pokemonNameWithAffix}} restored PP to its move {{moveName}}\nusing its {{berryName}}!",
+  "hpHealBerry": "{{pokemonNameWithAffix}} restored its health using\nits {{berryName}}!",
```
Comment (mercurius-00) [inline src/locales/zh_TW/battle.ts:58]:
```suggestion
  "ppHealBerry": "{{pokemonNameWithAffix}}用{{berryName}}\n回復了{{moveName}}的PP！",
  "hpHealBerry": "{{pokemonNameWithAffix}}用{{berryName}}\n回復了體力！",
```

**Review by mercurius-00 — COMMENTED** [review]
Zh-CN and Zh-TW

File: src/locales/fr/battle.ts [hunk 5]
```diff
@@ -66,6 +66,8 @@ export const battle: SimpleTranslationEntries = {
   "foePokemonWithAffix": "{{pokemonName}} ennemi",
   "useMove": "{{pokemonNameWithAffix}} utilise\n{{moveName}} !",
   "stealEatBerry": "{{pokemonName}} vole et mange\nla {{berryName}} de {{targetName}} !",
+  "ppHealBerry": "{{pokemonNameWithAffix}} restored PP to its move {{moveName}}\nusing its {{berryName}}!",
+  "hpHealBerry": "{{pokemonNameWithAffix}} restored its health using\nits {{berryName}}!",
```
Comment (Adri1) [inline src/locales/fr/battle.ts:70]:
```suggestion
  "ppHealBerry": "La {{berryName}} de {{pokemonNameWithAffix}}\nrestaure les PP de sa capacité {{moveName}} !",
  "hpHealBerry": "La {{berryName}} de {{pokemonNameWithAffix}}\nrestaure son énergie !",
```

**Review by Adri1 — COMMENTED** [review]
Added French

File: src/locales/pt_BR/battle.ts [hunk 6]
```diff
@@ -68,6 +68,8 @@ export const battle: SimpleTranslationEntries = {
   "drainMessage": "{{pokemonName}} teve sua\nenergia drenada!",
   "regainHealth": "{{pokemonName}} recuperou\npontos de saúde!",
   "stealEatBerry": "{{pokemonName}} stole and ate\n{{targetName}}'s {{berryName}}!",
+  "ppHealBerry": "{{pokemonNameWithAffix}} restored PP to its move {{moveName}}\nusing its {{berryName}}!",
+  "hpHealBerry": "{{pokemonNameWithAffix}} restored its health using\nits {{berryName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:72]:
```suggestion
  "hpHealBerry": "{{pokemonNameWithAffix}} restarou sua saúde usando\nsua {{berryName}}!",
```

File: src/locales/pt_BR/battle.ts [hunk 7]
```diff
@@ -68,6 +68,8 @@ export const battle: SimpleTranslationEntries = {
   "drainMessage": "{{pokemonName}} teve sua\nenergia drenada!",
   "regainHealth": "{{pokemonName}} recuperou\npontos de saúde!",
   "stealEatBerry": "{{pokemonName}} stole and ate\n{{targetName}}'s {{berryName}}!",
+  "ppHealBerry": "{{pokemonNameWithAffix}} restored PP to its move {{moveName}}\nusing its {{berryName}}!",
```
Comment (josericardo-fo) [inline src/locales/pt_BR/battle.ts:71]:
```suggestion
  "ppHealBerry": "{{pokemonNameWithAffix}} restaurou PP do movimento {{moveName}}\nusando sua {{berryName}}!",
```

**Review by CodeTappert — APPROVED** [review]
German is fine

**Review by josericardo-fo — APPROVED** [review]

**Review by Adri1 — APPROVED** [review]

**Comment by NicusPulcis** [comment]
hey @EnochG1 i've sent a PR with the italian translations

**Review by 40chyan — APPROVED** [review]

**Pull request merged**